test/spec/features/search/searchSpec.js

```diff
@@ -144,13 +144,52 @@ describe('features/search', function() {
     // then
     expect(results).to.have.length(5);
     expect(results[0].item).to.eql(items[1]);
-    expect(results[1].item).to.eql(items[5]);
-    expect(results[2].item).to.eql(items[0]);
+    expect(results[1].item).to.eql(items[0]);
+    expect(results[2].item).to.eql(items[5]);
     expect(results[3].item).to.eql(items[4]);
     expect(results[4].item).to.eql(items[3]);
   }));
 
 
+  it('should sort with all keys', inject(function(search) {
+
+    const items = [
+      {
+
+        id: 'resources-replace-form-0',
```

Comment:
I propose we simplify this example to make it easier to understand:

* You test that `f` and `form` matches items in `search`, we don't give up after comparing the label.
* Also needs testing:
  * An entry can be an array
  * Matches are prioritized top -> bottom (an earlier "start of word" match beats a later one)